### PR TITLE
Add txid field to the fundchannel call

### LINKED ISSUE #236: fundchannel should return a txid inside the json respose
We are missing the more important information that is the `txid`

```
➜  ~ lampo-cli --network signet fundchannel --node_id 026b92e6cc3e0f24b42494672ecf65f3d147b9deb95186bf024da7a7745008ee75  --amount 200000 --public true
{
  "node_id": "026b92e6cc3e0f24b42494672ecf65f3d147b9deb95186bf024da7a7745008ee75",
  "amount": 200000,
  "public": true,
  "push_mst": 0,
  "to_self_delay": 2016,
  "tx": {
    "version": 2,
    "lock_time": 0,
    "input": [
      {
        "previous_output": "ac9ee361e0ec07f919f6b9dc07c2fd21f05ef559326b45e6f889c6e37bb6f32b:0",
        "script_sig": "",
        "sequence": 4294967294,
        "witness": [
          "55846fb5c7ef20f0ba055a8cc9c3d9e97c4323d02bbf9b9a88490d4fd24cacde5afeac503ca26a163089c3b3a4246340cfffa2dcc7388c5055d139871e101449"
        ]
      },
      {
        "previous_output": "9891b12bb6e608bada7296bdc47bd424df16ad4ac14330c0f003335df18b9b03:0",
        "script_sig": "",
        "sequence": 4294967294,
        "witness": [
          "f6e0d011c6052ccf697ed5a5c48cee62acf5f2aed548bf27d71b31951072b5ef150a49897ec260a0c80dc4246903affb62bb21ad80036b2a0ec551c3b1ccc6fa"
        ]
      }
    ],
    "output": [
      {
        "value": 1059880,
        "script_pubkey": "51209a9177c972471352f58c4b9b8fedcaea9b0230edc1565c671644119e84059caf"
      },
      {
        "value": 200000,
        "script_pubkey": "0020ad50c2d4bbcc6db44e70bdf30acc123bd5b62652df250bd7fb29641d8083cd70"
      }
    ]
  }
}
```

### PR DESCRIPTION
**Pull request by chrisguida**
Fixes #236
Screenshot from 2024-06-18 13-48-17

### CODE REVIEW

**Comment by vincenzopalazzo** [comment]
Woww! thanks <3! 

Will look tomorrow morning

File: lampod/src/ln/channel_manager.rs [hunk 1]
```diff
@@ -432,13 +432,19 @@ impl ChannelEvents for LampoChannelManager {
...
         Ok(response::OpenChannel {
             node_id: open_channel.node_id,
             amount: open_channel.amount,
             public: open_channel.public,
             push_mst: 0,
             to_self_delay: 2016,
             tx,
+            txid,
         })
     }
```
Comment (vincenzopalazzo) [inline lampod/src/ln/channel_manager.rs:449]: it is possible write this match in a single line?

A possible diff can be 

```diff
diff --git a/lampod/src/ln/channel_manager.rs b/lampod/src/ln/channel_manager.rs
index f65f67c..ab6f83d 100644
--- a/lampod/src/ln/channel_manager.rs
+++ b/lampod/src/ln/channel_manager.rs
@@ -432,11 +432,7 @@ impl ChannelEvents for LampoChannelManager {
             }
         };

-        let txid = match &tx {
-            Some(t) => Some(t.txid()),
-            None => None,
-        };
-
+        let txid = tx.as_ref().map(|tx| tx.txid());
         Ok(response::OpenChannel {
             node_id: open_channel.node_id,
             amount: open_channel.amount,
```

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]
Just a single comment, otherwise LGTM

File: lampod/src/ln/channel_manager.rs (hunk 1 above)
Comment (chrisguida) [reply lampod/src/ln/channel_manager.rs]: Yep, your way is better. Updated.

**Review by vincenzopalazzo — APPROVED** [review]
ACK https://github.com/vincenzopalazzo/lampo.rs/pull/257/commits/1772b06ff7ad1d56d5941b65910ca424ea5f8670

**Pull request merged**